Client/Protocols/Presenter.swift

```diff
@@ -0,0 +1,22 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/// Protocol to present a view controller
+protocol Presenter {
```

Comment:
I agree and thanks for sharing 🙌 
Whilst cherry-picking I had to solve conflicts and hence took the chance to not commit and look at the changes.
This `Presenter` sounded 🐟 but, I eventually gone after your exact last thought 💭 👇 
> I'm guessing in the end it might make sense to keep it if not just to be closer to FFs code, but wanted to share my thoughts out loud if anyone had anything to add here.

Reply:
I'd say let's keep this protocol though it's only used once so far in order to not diverge too much from FF. I think it's used in other places in the FF codebase already and thus will make more sense to have after the next upgrade.